server/ec2spotmanager/models.py

```diff
@@ -32,32 +33,92 @@ def get_available_name(self, name, max_length=None):
             os.remove(os.path.join(getattr(settings, 'USERDATA_STORAGE', None), name))
         return name
 
-
 class PoolConfiguration(models.Model):
```

Comment:
Model-based inheritance has a number of downsides that we discussed when designing the provider abstraction interface. I don't see why we need the complication it adds when the abstraction is done in `CloudProvider` instead. Having all fields in PoolConfiguration also allows a single pool to support multiple providers in the future, which would be really complicated with model-based inheritance.

Reply:
the model based inheritance is removed.